src/main/java/frc/robot/subsytems/Drivetrain/DriveImpl.java

```diff
@@ -2,22 +2,30 @@
 
 import java.util.concurrent.TimeUnit;
 
+import ca.team3161.lib.robot.motion.drivetrains.SpeedControllerGroup;
 import ca.team3161.lib.robot.subsystem.RepeatingPooledSubsystem;
-import edu.wpi.first.wpilibj.SpeedController;
+import edu.wpi.first.wpilibj.Encoder;
 import edu.wpi.first.wpilibj.drive.DifferentialDrive;
 import edu.wpi.first.wpiutil.math.Pair;
+import frc.robot.RobotMap;
 
 public class DriveImpl extends RepeatingPooledSubsystem implements Drive {
+    
+    
 
-    private final SpeedController leftSide;
-    private final SpeedController rightSide;
-    private final DifferentialDrive drivetrain;
+    SpeedControllerGroup leftSide;
+    SpeedControllerGroup rightSide;
+    DifferentialDrive drivetrain;
+    Encoder leftEncoder;
+    Encoder rightEncoder;
 
-    public DriveImpl(SpeedController leftSide, SpeedController rightSide) {
+    public DriveImpl(SpeedControllerGroup leftSide, SpeedControllerGroup rightSide) {
         super(20, TimeUnit.MILLISECONDS);
         this.leftSide = leftSide;
         this.rightSide = rightSide;
         this.drivetrain = new DifferentialDrive(leftSide, rightSide);
+        this.leftEncoder = new Encoder(RobotMap.LEFT_ENCODER_PORTS[0], RobotMap.LEFT_ENCODER_PORTS[1], false, Encoder.EncodingType.k2X);
+        this.rightEncoder = new Encoder(RobotMap.RIGHT_ENCODER_PORTS[0], RobotMap.RIGHT_ENCODER_PORTS[1], false, Encoder.EncodingType.k2X);
```

Comment:
Just for the sake of following the pattern, let's pass the Encoders into the constructor the same way that is done with the SpeedControllerGroups. So the Encoders will actually be instantiated in Robot.java.